### PR TITLE
Autodetect platform using domain regex

### LINKED ISSUE #243: Autodetect platform using domain regex
Platforms like `Bitbucket Server` and `Gerrit` do not have a predefined domain as `github.com` .
Because of that, the plugin never autodetects these kinds of platform on new cloned projects.

It would be nice to be able to provide a domain regex for such platforms.

For example, for `Gerrit` platform, the domain regex `.*gerrit.*` would allow to match `gerrit.mycompany.com` thus would free mycompany’s users from manually choosing Gerrit as their platform for each `mycompany` cloned project.

### PR DESCRIPTION
**Pull request by reda-alaoui**
Fixes #243

### CODE REVIEW

**Comment by reda-alaoui** [comment]
Ping :)

**Comment by ben-gibson** [comment]
I'll take a look at this one later tonight.

File: src/main/kotlin/uk/co/ben_gibson/git/link/platform/Platform.kt [hunk 1]
```diff
@@ -48,7 +49,9 @@ class BitbucketCloud() : Platform(
 class BitbucketServer() : Platform(
     UUID.fromString("dba5941d-821c-49b3-83b0-75deb9462acb"),
     message("platform.bitbucket.server.name"),
-    Icons.BITBUCKET
+    Icons.BITBUCKET,
+    setOf(),
+    Pattern.compile(".*bitbucket.*", Pattern.CASE_INSENSITIVE)
```
Comment (ben-gibson) [inline src/main/kotlin/uk/co/ben_gibson/git/link/platform/Platform.kt:54]: Does this have any impact on the `BitbucketCloud` detection? Or does that get checked first and match `bitbucket.org` before reaching this check

Comment (reda-alaoui) [reply src/main/kotlin/uk/co/ben_gibson/git/link/platform/Platform.kt:54]: No impact.
Domain patterns are checked in a loop coming after domain loop.

**Comment by huyz** [comment]
how does this feature work? can we get a screenshot?

**Comment by reda-alaoui** [comment]
The user has nothing to do. There is no screenshot to provide IMO.

- If your `Gerrit` instance's domain (e.g. `gerrit.company.com`) **contains** the word `gerrit` with any case, `GitLink` will autodetect the `Gerrit` platform.
- If your `Bitbucket Server` instance's domain (e.g. `bitbucket.company.com`) **contains** the word `bitbucket` with any case, `GitLink` will autodetect the `Bitbucket Server` platform.

**Comment by huyz** [comment]
Oh so this doesn't allow for mapping custom regular expressions to the provider we want?

**Pull request merged**